projects/storefrontlib/src/cms-components/misc/qualtrics/qualtrics-loader.service.ts

```diff
@@ -0,0 +1,76 @@
+import { Injectable } from '@angular/core';
+import { WindowRef } from '@spartacus/core';
+import { BehaviorSubject, fromEvent, Observable, of } from 'rxjs';
+import { distinctUntilChanged, filter, mergeMap, tap } from 'rxjs/operators';
+import { QualtricsConfig } from '../../../shared/config/qualtrics-config';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class QualtricsLoaderService {
+  qualtricsLoaded$ = new BehaviorSubject<boolean>(false);
+
+  constructor(private winRef: WindowRef, private config: QualtricsConfig) {
+    if (Boolean(this.winRef.nativeWindow)) {
+      this.initialize();
+      this.setup();
+    }
+  }
+
+  load(): Observable<boolean> {
+    return this.qualtricsLoaded$.pipe(
+      filter(loaded => loaded),
+      mergeMap(_ => {
```

Comment:
What do you think, which one should we use here?
- If we use `concatMap` every call to `load()` will be queued.
- If we use `switchMap`, the previous call will be cancelled in favour to the newer call.
- `mergeMap` will run every call in parallel.
- `exhaustMap` will drop any newer call if the current one is still processing.

Reply:
Thanks for picking this out, I think I rather use switchMap, because we want the latest and greatest call instead of running things in parallel